### ISSUE TITLE
TripletDataGenerator class does not utilize PyTorch data loading utilities and lacks data shuffling

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#352]
The TripletDataGenerator class does not properly utilize PyTorch's data loading utilities and lacks the ability to shuffle data at the beginning of each epoch, potentially leading to overfitting and inefficient training. The class also does not increment epoch number properly.
